### PR TITLE
Fix memory leak in ResumptionTicketTest

### PR DESCRIPTION
**Pull request by anrossi**

### CODE REVIEW

File: src/core/unittest/TicketTest.cpp [hunk 1]
```diff
@@ -103,6 +103,26 @@ struct TransportParamsScope {
...
 
+struct TicketScope {
+    uint8_t* p = nullptr;
+
+    ~TicketScope() {
+        reset();
+    }
+
+    uint8_t** operator&() {
+        reset();
```
Comment (nibanks) [inline src/core/unittest/TicketTest.cpp:114]: Why the reset here? I don't like this hidden side effect.

Comment (anrossi) [reply src/core/unittest/TicketTest.cpp:114]: I believe `wil::unique_ptr` uses the same contract, to prevent leaked memory. But I'll change it up.

Comment (ThadHouse) [reply src/core/unittest/TicketTest.cpp:114]: ComPtr uses a GetAddressOf() function to give this behavior and then just an operator uint8_t* to get the value without reset if necessary. That'd probably be what I'd do.

Comment (anrossi) [reply src/core/unittest/TicketTest.cpp:114]: Confirmed this is the way `wistd::unique_any_t` works, which is a base-class for `wistd::unique_ptr`: https://github.com/microsoft/wil/blob/ac455d2c2e3018dc37aafcdf0ffcc17adcd5739b/include/wil/resource.h#L319

File: src/core/unittest/TicketTest.cpp [hunk 2]
```diff
@@ -103,6 +103,28 @@ struct TransportParamsScope {
...
 };
 
+struct TicketScope {
+    uint8_t* p = nullptr;
+
+    ~TicketScope() {
+        reset();
+    }
+
+    uint8_t** operator&() = delete;
```
Comment (ThadHouse) [inline src/core/unittest/TicketTest.cpp:113]: This operator isn't defined anyway, so there is no reason to delete it. the default operator& will return a TicketScope*.

Comment (anrossi) [reply src/core/unittest/TicketTest.cpp:113]: Why not explicitly discourage the usage of `(uint8_t**) operator&()`? Won't that be covered by this?

**Review by nibanks — APPROVED** [review]

**Pull request merged**